py-polars/polars/dataframe/frame.py

```diff
@@ -8283,6 +8298,162 @@ def rows(
         else:
             return self._df.row_tuples()
 
+    def rows_by_key(
+        self,
+        key: str | Sequence[str] | SelectorType,
+        *,
+        named: bool = False,
+        include_key: bool = False,
+        unique: bool = False,
+    ) -> dict[Any, Iterable[Any]]:
+        """
+        Returns DataFrame data as a keyed dictionary of python-native values.
```

Comment:
Can we add a note about compute cost of materializing? Maybe explain that this should be used to get data out of polars -> python, but not to do any serious computing that can be done within the polars engine itself.

Reply:
Will do 👍